k_tree/k_tree.py

```diff
@@ -29,82 +27,82 @@ def __repr__(self):
     def __str__(self):
         return str(self.root.val)
 
-    def pre_order(self, opertion):
+    def pre_order(self, operation):
+        """pre-order traversal of a k-ary tree"""
         def _walk(node=None):
             if node is None:
                 return
 
             operation(node)
 
-            for child in node.children:
-                _walk(child)
+            if len(node.children):
```

Comment:
What you had before was fine. It is not necessary to verify if the `node` has children since a loop will not execute if there is nothing in the list.